variables.tf

```diff
@@ -214,14 +220,118 @@ variable "worker_cpu_cores" {
   type        = string
   default     = "eth0"
 }
+
+variable "proxmox_api_token" {
+  description = "Proxmox API token in format 'user@realm!tokenname=token-secret'"
+  type        = string
+  sensitive   = true
+}
+
+variable "proxmox_ssh_agent" {
+  description = "Use SSH agent for Proxmox SSH connection"
+  type        = bool
+  default     = false
+}
+
+variable "proxmox_ssh_username" {
+  description = "SSH username for Proxmox connection"
+  type        = string
+  default     = "root"
+}
+
+variable "proxmox_ssh_private_key_path" {
+  description = "Path to SSH private key for Proxmox connection"
+  type        = string
+}
+
+
+variable "network_bridge" {
+  description = "Network bridge for VM network interfaces"
+  type        = string
+  default     = "vmbr0"
+}
+
+# VM Template IDs
+variable "controlplane_template_id" {
+  description = "VM ID for the control plane template"
+  type        = number
+  default     = 998
+  validation {
+    condition     = var.controlplane_template_id > 0 && var.controlplane_template_id < 10000
+    error_message = "Control plane template ID must be between 1 and 9999."
+  }
+}
+
+variable "worker_template_id" {
+  description = "VM ID for the worker template"
+  type        = number
+  default     = 999
+  validation {
+    condition     = var.worker_template_id > 0 && var.worker_template_id < 10000
+    error_message = "Worker template ID must be between 1 and 9999."
+  }
+}
+
+# VM Starting IDs
+variable "controlplane_vm_id_start" {
+  description = "Starting VM ID for control plane nodes"
+  type        = number
+  default     = 2000
+  validation {
+    condition     = var.controlplane_vm_id_start > 0 && var.controlplane_vm_id_start < 10000
+    error_message = "Control plane VM ID start must be between 1 and 9999."
+  }
+}
+
+variable "worker_vm_id_start" {
+  description = "Starting VM ID for worker nodes"
+  type        = number
+  default     = 3000
+  validation {
+    condition     = var.worker_vm_id_start > 0 && var.worker_vm_id_start < 10000
+    error_message = "Worker VM ID start must be between 1 and 9999."
+  }
+}
+
+# Disk Sizes (in GB)
+variable "controlplane_disk_size" {
+  description = "Disk size for control plane nodes in GB"
+  type        = number
+  default     = 20
+  validation {
+    condition     = var.controlplane_disk_size >= 10
+    error_message = "Control plane disk size must be at least 10 GB."
+  }
+}
+
+variable "worker_disk_size" {
+  description = "Disk size for worker nodes in GB"
+  type        = number
+  default     = 50
+  validation {
+    condition     = var.worker_disk_size >= 20
+    error_message = "Worker disk size must be at least 20 GB."
+  }
+}
+
+variable "cloudinit_disk_size" {
+  description = "Cloud-init disk size in MB"
+  type        = number
+  default     = 4
+  validation {
+    condition     = var.cloudinit_disk_size >= 1
+    error_message = "Cloud-init disk size must be at least 1 MB."
+  }
+}
```

Comment:
```suggestion
```
Cloud init removed in favor of [talos_machine_configuration_apply](https://registry.terraform.io/providers/siderolabs/talos/latest/docs/resources/machine_configuration_apply)